### DISCUSSION TITLE
JSON Schema output format for validation results.

### DISCUSSION THREAD

**Discussion by jugaadi** [cue-lang/cue#1773] [General]
Any plans to support JSON Schema output format like its mentioned here: https://json-schema.org/draft/2020-12/json-schema-core.html#rfc.section.12  or its equivalent for validation results?

**Comment by slewiskelly** [comment]
[This issue](https://github.com/cue-lang/cue/issues/929) covers it?

**Reply by verdverm** [reply]
I'm not sure this answers the question. My reading was that JSONSchema looks to have validation output as a JSON object, and the question is whether CUE might support outputting validation errors in a machine readable format for secondary processing and presentation. The linked issue is for he JSONSchema codec, whereas the structured validation errors output could seems more generalized.

@jugaadi can you confirm?

**Reply by jugaadi** [reply]
Yes. We would need errors as well as success results for processing annotations.

**Reply by verdverm** [reply]
Does it need to follow that spec? That seems to talk about things specific to JSONSchema.

I'd be interested to see CUE have an `--errorFormat` flag that supports text (current / default), json, yaml, jsonschema, etc (formats that CUE supports for export)

**Comment by rogpeppe** [comment]
@jugaadi This is an interesting question, thanks! I'm pretty sure it doesn't support anything like this right now. Perhaps worth adding a feature-request issue if you want this feature (and if you do, please also say what your context is and why the feature would be specifically useful to you)
